src/runtime_src/core/pcie/emulation/hw_em/alveo_shim/plugin/xdp/pl_deadlock.cpp

```diff
@@ -0,0 +1,65 @@
+/**
+ * Copyright (C) 2021 Xilinx, Inc
+ * Copyright (C) 2022 Advanced Micro Devices, Inc. - All rights reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may
+ * not use this file except in compliance with the License. A copy of the
+ * License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
```

Comment:
It would be nice to have in any file some useful comments explaining what this file is for.
Some design ideas are always welcome!

Reply:
@keryell pretty much all of XRT follow this plug and play design. As different things need to be loaded for hw, hw emu, windows, edge, aie etc.